File: src/main/java/edu/kit/ifv/mobitopp/actitopp/DCAlternative.java
package edu.kit.ifv.mobitopp.actitopp;

/**
 * 
 * @author Tim Hilgert
 *
 */
public class DCAlternative
{
  private UtilityFunction utilityFunction;

  
  private String name;
  private double probability=-1;
  private boolean enabled = true;
  
  /*
   * Faktor to weight utility.
   * by default, factor is equal to 1, i.e. no additional
   * weithing. Some model steps used weighting factor to ensure
   * stability aspects, i.e. setting it to 1.1 is equal to a 10%
   * raising of the utility.
   * 
   * factor is inclued when getUtility
   */
  private double utilityweithingfactor=1.0;
  
  /**
   * 
   * Konstruktor
   * 
   * @param name
   */
	public DCAlternative(String name)
	{
	  super();
	  this.name = name;
	  utilityFunction = new UtilityFunction();
	}

	public UtilityFunction getUtilityFunction()
  {
    return utilityFunction;
  }

  /**
	 * @param utilityfactor the utilityfactor to set
	 */
	public void setUtilityfactor(double utilityfactor) {
		this.utilityweithingfactor = utilityfactor;
	}

	public String getName()
  {
    return name;
  }

  public void setName(String name)
  {
    this.name = name;
  }

	/**
	 * @return the utility*utilityFactor
	 */
  public double getUtility()
  {
    return utilityFunction.getUtility() * utilityweithingfactor;
  }
  
	/**
	 * @return the probability
	 */
	public double getProbability() {
		return probability;
	}

	/**
	 * @param probability the probability to set
	 */
	public void setProbability(double probability) {
		this.probability = probability;
	}

	/**
	 * @return the enabled
	 */
	public boolean isEnabled() {
		return enabled;
	}

	/**
	 * @param enabled the enabled to set
	 */
	public void setEnabled(boolean enabled) {
		this.enabled = enabled;
	}

	public String toString()
  {
  	return getName();
  }
	
}

File: src/main/java/edu/kit/ifv/mobitopp/actitopp/DCAlternative.kt
package edu.kit.ifv.mobitopp.actitopp

import edu.kit.ifv.mobitopp.actitopp.utilityFunctions.deprecated.UtilityFunction

/**
 * @author Tim Hilgert
 */
class DCAlternative( var name: String) {
    
    val utilityFunction: UtilityFunction = UtilityFunction()


    /**
     * @return the probability
     */
    /**
     * @param probability the probability to set
     */
    
    var probability: Double = -1.0
    /**
     * @return the enabled
     */
    /**
     * @param enabled the enabled to set
     */
    var isEnabled: Boolean = true

    /*
     * Faktor to weight utility.
     * by default, factor is equal to 1, i.e. no additional
     * weithing. Some model steps used weighting factor to ensure
     * stability aspects, i.e. setting it to 1.1 is equal to a 10%
     * raising of the utility.
     *
     * factor is inclued when getUtility
     */
    private var utilityweithingfactor = 1.0

    /**
     * @param utilityfactor the utilityfactor to set
     */
    fun setUtilityfactor(utilityfactor: Double) {
        this.utilityweithingfactor = utilityfactor
    }

    val utility: Double
        /**
         * @return the utility*utilityFactor
         */
        get() = utilityFunction.utility * utilityweithingfactor

    override fun toString(): String {
        return name
    }
}
